### ISSUE TITLE
Prettier-atom doesn't work when prettier install on root package.json of yarn workspaces

### ISSUE THREAD

**Issue by oriSomething** [prettier/prettier-atom#418]
`prettier-atom` doesn't work when `prettier` install in root `package.json` of workspaces.
- It works with `prettier-vscode`
- If I manually run the `prettier` command of the root of workspaces from CLI, It works
- If I remove `prettier` dependency from the root `package.json` of "yarn workspace", it works

`Prettier: Debug` output:

```
Atom version: 1.27.1
prettier-atom version: 0.53.0
prettier: bundled
prettier version: 1.11.1
prettier-eslint version: 8.8.1
prettier-atom configuration: {
  "formatOnSave": true,
  "formatOnSaveOptions": {
    "enabled": true,
    "excludedGlobs": [
      "package.json",
      "index.html",
      "*.md",
      "*.flow"
    ],
    "showInStatusBar": true,
    "respectEslintignore": true,
    "whitelistedGlobs": [],
    "isDisabledIfNotInPackageJson": false,
    "isDisabledIfNoConfigFile": false
  },
  "formatOnSaveSilentError": true,
  "jsxBracketSameLine": true,
  "prettierOptions": {
    "jsxBracketSameLine": true,
    "printWidth": 120,
    "trailingComma": "all",
    "singleQuote": false,
    "bracketSpacing": true,
    "semi": true,
    "useTabs": false,
    "tabWidth": "auto",
    "parser": "babylon",
    "arrowParens": "avoid"
  },
  "silenceErrors": false,
  "useFlowParser": false,
  "useEslint": false,
  "useStylelint": false,
  "useEditorConfig": true,
  "prettierEslintOptions": {
    "prettierLast": false
  },
  "scopes": {
    "javascript": [
      "source.js",
      "source.jsx",
      "source.js.jsx",
      "source.babel",
      "source.js-semantic",
      "text.html.basic",
      "text.html.vue"
    ],
    "typescript": [
      "source.ts",
      "source.tsx",
      "source.ts.tsx"
    ],
    "css": [
      "source.css",
      "source.less",
      "source.css.less",
      "source.scss",
      "source.css.scss",
      "source.css.postcss"
    ],
    "json": [
      "source.json"
    ],
    "graphQl": [
      "source.graphql"
    ],
    "markdown": [
      "source.md",
      "source.gfm",
      "text.md"
    ],
    "vue": [
      "text.html.vue"
    ]
  }
}
```

The error stack trace in Atom:
```
[ERROR] atom-error-notification -  prettier-atom failed: No parser and no file path given, couldn't infer a parser.
Error: prettier-atom failed: No parser and no file path given, couldn't infer a parser.
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at NotificationManager.addNotification (/Applications/Atom.app/Contents/Resources/app/src/notification-manager.js:193:24)
    at NotificationManager.atom.notifications.addNotification.notification [as addNotification] (/Users/ori/.atom/packages/nuclide/lib/main.js:360:52)
    at NotificationManager.addError (/Applications/Atom.app/Contents/Resources/app/src/notification-manager.js:154:23)
    at addErrorNotification (/Users/ori/.atom/packages/prettier-atom/dist/atomInterface/index.js:143:29)
    at displayErrorInPopup (/Users/ori/.atom/packages/prettier-atom/dist/executePrettier/handleError.js:56:3)
    at n (/Users/ori/.atom/packages/prettier-atom/node_modules/lodash/lodash.min.js:5:89)
    at /Users/ori/.atom/packages/prettier-atom/node_modules/lodash/lodash.min.js:97:428
    at n (/Users/ori/.atom/packages/prettier-atom/node_modules/lodash/lodash.min.js:5:89)
    at /Users/ori/.atom/packages/prettier-atom/node_modules/lodash/lodash.min.js:63:257
    at /Users/ori/.atom/packages/prettier-atom/node_modules/lodash/lodash.min.js:49:114
    at _callee2$ (/Users/ori/.atom/packages/prettier-atom/dist/executePrettier/executePrettierOnBufferRange.js:175:13)
    at tryCatch (/Users/ori/.atom/packages/prettier-atom/node_modules/regenerator-runtime/runtime.js:62:40)
    at Generator.invoke [as _invoke] (/Users/ori/.atom/packages/prettier-atom/node_modules/regenerator-runtime/runtime.js:296:22)
    at Generator.prototype.(anonymous function) [as throw] (/Users/ori/.atom/packages/prettier-atom/node_modules/regenerator-runtime/runtime.js:114:21)
    at step (/Users/ori/.atom/packages/prettier-atom/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
    at /Users/ori/.atom/packages/prettier-atom/node_modules/babel-runtime/helpers/asyncToGenerator.js:30:13
    at <anonymous>
```

**Comment by oriSomething** [comment]
From investigating, adding `parser: "babylon"` (can be `flow` too) field to `prettier.config.js` solves it.
It's related to the fact that the package `prettier` itself, calls `format` without `parser` options. Currently, not sure why there was no auto infer of the parser from `prettier-atom` perspective.

**Comment by astorije** [comment]
I am hitting a similar error and wondering if it has to do with: https://prettier.io/blog/2018/05/27/1.13.0.html#don-t-default-to-the-javascript-parser-4528-by-duailibe

**Comment by oriSomething** [comment]
@astorije I don't think. Because as I said, if I remove the `prettier` dependency from the workspaces. `prettier-atom` formats as usual

**Comment by astorije** [comment]
`prettier-atom` relies on Prettier 1.11.something if you don't have one in your dependencies. What version of Prettier do you have in your dependencies?

Btw with 1.13.2 this _might_ be better, as they reverted some changes to the default parser thingy.

**Comment by oriSomething** [comment]
`1.13.0`

**Comment by astorije** [comment]
Yep, I confirm I could hit the error with 1.13.0, and it is fixed with 1.13.2.

**Comment by oriSomething** [comment]
I've checked it now. It seems the issue was fixed
